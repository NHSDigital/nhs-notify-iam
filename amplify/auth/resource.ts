import { DeepPartialAmplifyGeneratedConfigs } from '@aws-amplify/plugin-types';
import { ClientConfig } from '@aws-amplify/client-config';
import { defineAuth } from '@aws-amplify/backend';

export const remoteAuthConfig: DeepPartialAmplifyGeneratedConfigs<ClientConfig> =
  {
    auth: {
      aws_region: 'eu-west-2',
      user_pool_id: process.env.USER_POOL_ID,
      user_pool_client_id: process.env.USER_POOL_CLIENT_ID,
    },
  };

export const sandboxAuthConfig = {
  auth: defineAuth({
    loginWith: {
      email: true,
    },
  }),
};
