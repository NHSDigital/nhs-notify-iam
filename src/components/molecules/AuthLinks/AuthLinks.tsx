'use client';

import React from 'react';
import { useAuthenticator } from '@aws-amplify/ui-react';
import { getConstants } from '@/src/utils/public-constants';
import content from '@/src/content/content';

const { BASE_PATH } = getConstants();

export const AuthLinks = () => {
  const { authStatus } = useAuthenticator();

  if (authStatus === 'authenticated') {
    return (
      <a
        id='sign-out-link'
        className='nhsuk-account__sign-in--link'
        href={`${BASE_PATH}/signout`}
      >
        {content.components.headerComponent.links.signOut}
      </a>
    );
  }

  return (
    <a
      id='sign-in-link'
      className='nhsuk-account__sign-in--link'
      href={BASE_PATH}
    >
      {content.components.headerComponent.links.signIn}
    </a>
  );
};
