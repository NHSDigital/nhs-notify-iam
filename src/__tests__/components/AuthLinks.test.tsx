import { mockDeep } from 'jest-mock-extended';
import { UseAuthenticator, useAuthenticator } from '@aws-amplify/ui-react';
import { render, screen } from '@testing-library/react';
import { AuthSession } from '@aws-amplify/auth';
import { AuthLinks } from '@/src/components/molecules/AuthLinks/AuthLinks';

jest.mock('@aws-amplify/ui-react');
jest.mock('aws-amplify/auth', () => ({
  fetchAuthSession: async () =>
    mockDeep<AuthSession>({
      tokens: {
        idToken: {
          payload: {
            email: 'test-email',
          },
        },
      },
    }),
}));

const useAuthenticatorMock = jest.mocked(useAuthenticator);

test('AuthLinks - authenticated', async () => {
  useAuthenticatorMock.mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'authenticated',
    })
  );

  const container = render(<AuthLinks />);

  await screen.findByText('Sign out');

  expect(container.asFragment()).toMatchSnapshot();
});

test('AuthLinks - unauthenticated', async () => {
  useAuthenticatorMock.mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'unauthenticated',
    })
  );

  const container = render(<AuthLinks />);

  await screen.findByText('Sign in');

  expect(container.asFragment()).toMatchSnapshot();
});

test('AuthLinks - configuring', async () => {
  jest.mocked(useAuthenticator).mockReturnValue(
    mockDeep<UseAuthenticator>({
      authStatus: 'configuring',
    })
  );

  const container = render(<AuthLinks />);

  await screen.findByText('Sign in');

  expect(container.asFragment()).toMatchSnapshot();
});
