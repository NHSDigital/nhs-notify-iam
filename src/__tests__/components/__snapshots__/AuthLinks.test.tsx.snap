// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthLinks - authenticated 1`] = `
<DocumentFragment>
  <a
    class="nhsuk-account__sign-in--link"
    href="/auth/signout"
    id="sign-out-link"
  >
    Sign out
  </a>
</DocumentFragment>
`;

exports[`AuthLinks - configuring 1`] = `
<DocumentFragment>
  <a
    class="nhsuk-account__sign-in--link"
    href="/auth"
    id="sign-in-link"
  >
    Sign in
  </a>
</DocumentFragment>
`;

exports[`AuthLinks - unauthenticated 1`] = `
<DocumentFragment>
  <a
    class="nhsuk-account__sign-in--link"
    href="/auth"
    id="sign-in-link"
  >
    Sign in
  </a>
</DocumentFragment>
`;
